HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+
+This HIP changes this behavior in the following ways:
+
+- The value `-1` is now permitted for `max_auto_associations`, and this value means “unlimited”.
+- The account holder no longer prepays for unused slots regardless of the setting for `max_auto_associations`. Slots are only paid for when used, and are initially paid for by the sender during automatic associations. The account is responsible for covering rent for each association after the first auto-renewal period. `used_auto_associations` is incremented every time an auto-association is made.
+- The default value for used_auto_associations for new automatically-created accounts will now be -1. This means, if an account is created automatically (by performing a token transfer to an alias that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured with unlimited automatic token associations. Accounts created by using the [AccountCreateTransaction](https://docs.hedera.com/hedera/sdks-and-apis/sdks/accounts-and-hbar/create-an-account) (HAPI: [CryptoCreate](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto))
+will continue to have a used_auto_associations default of 0. Accounts created before this feature is released will remain unchanged.
+
+### Crypto Transfers
+
+Normal crypto-transfers remain the same as today, with one difference. If Account A sends a token to Account B, and Account B is not associated with that token but has an available auto-association slot, then Account A will pay for that association and the first auto-renewal period’s rent, in addition to the typical transfer costs. In summary previously the reciever would pay for an association in advance, not the payer of the transaction.
+
+![Crypto Transfer Transaction](../assets/hip-904/frictionless-airdrop-cryptoTransferTransaction.png)
+
+### Airdrops
+
+Airdrops are similar to crypto transfers, but differ in one fundamental way - when an airdrop is performed and the receiver does not have available or automatic association slots, rather than failing, the transfer will be kept in state as a *pending transfer*. To represent this new semantic, there will be a new `TokenAirdropTransaction` meant for use specifically by those attempting an airdrop, with the following semantics:
```

Comment:
If we have a new type `TokenAirdropTransaction`, shouldn't we perform the new economics of paying for the token association and the first rental period only in this case. This would keep the `CryptoTransfers` the same and won't increase their fee. In my opinion, we should strive to achieve as low fees as possible for all type of operations.

Reply:
I'd agree with you if it weren't a problem today that receivers had to pay for associations in advance for tokens they may or may not receive.
Rather the fee is the same but it's moved from receiver to payer when it's association.

The new first rental period is to allow receivers enough time to recognize their automatic association and act upon it.
By sending token value to an account that doesn't already have balance a sender is initiating the balance and takes responsibility for that.

Reply:
To be clear. A token airdrop that results in a pending state will charge the sender an association fee in advance of the token ever actually making it to the receiver.